### PR TITLE
 Exclude Beast WebSocket from build.

### PR DESCRIPTION
**Pull request by JunielKatarn**
Beast-based WebSocket implementation (`BeastWebSocketResource.cpp`) has been deprecated and should be excluded from the Desktop DLL.

Keeping the code, which can be enabled in the build using MSBuild property `EnableBeast` to conduct pending performance comparisons with the WinRT implementation.

The immediate benefit is dropping ~3MB in size from `react-native-win32.dll` (the bulk of it being OpenSSL symbols).

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6556)

### CODE REVIEW

**Comment by NickGerleman** [comment]
Does this mean we can also remove the ReactWindows.OpenSSL.StdCall.Static consumption?

Would be great to be able to eventually properly remove the code. Sounds like something we would do after the perf testing you were wanting to do?

**Comment by JunielKatarn** [comment]
> Does this mean we can also remove the ReactWindows.OpenSSL.StdCall.Static consumption?
> 
> Would be great to be able to eventually properly remove the code. Sounds like something we would do after the perf testing you were wanting to do?

It certainly does, but only for the Desktop (production) DLL. I would hold off such change until we actually delete the Beast code.
For now, the build process will just not use the dependency for that artifact, but OpenSSL is used bye integration test code.

File: vnext/PropertySheets/React.Cpp.props [hunk 1]
```diff
@@ -21,6 +21,7 @@
   <PropertyGroup Label="Desktop">
     <!-- See https://docs.microsoft.com/en-us/cpp/porting/modifying-winver-and-win32-winnt -->
     <WinVer>_WIN32_WINNT_WIN8</WinVer>
+    <EnableBeast Condition="'$(EnableBeast)' == ''">0</EnableBeast>
```
Comment (NickGerleman) [inline vnext/PropertySheets/React.Cpp.props:24]: Nit: idiom is to test against the string "true" instead of using numbers

**Review by NickGerleman — APPROVED** [review]

File: vnext/PropertySheets/React.Cpp.props (hunk 1 above)
Comment (JunielKatarn) [reply vnext/PropertySheets/React.Cpp.props:24]: I considered it, but the value would need to be translated again into `1` or `0` to use in the C++ macro `ENABLE_BEAST` (via additional `ItemDefinitionGroup` entry).
For a transitory property, I think it should be fine to use an unorthodox convention.

**Review by dannyvv — APPROVED** [review]

**Pull request merged**